### ISSUE TITLE
report on area (m2) of modelled habitat 

### ISSUE THREAD

**Issue by NewGraphEnvironment** [smnorris/bcfishpass#196]
Although our ability to [predict channel width](https://www.poissonconsulting.ca/temporary-hidden-link/859859031/channel-width-21b/) based on a model of discharge built from watershed area and mean annual precipitation estimates weighted by upstream watershed area is currently somewhat coarse it appears to be improving, not super duper poor and can be [supplemented by actual measured data](https://github.com/smnorris/bcfishpass/blob/main/scripts/channel_width/sql/channel_width_measured.sql).

Currently stream species modelling includes only linear lengths of streamline modelled upstream of specific points (stream crossing locations).  As we now have an estimate of channel width for each >1st order stream in the freshwater atlas that lands fully within BC, presenting estimates of area upstream is feasible.  

An example of why this would be so helpful is comparison of two streams with 1km of habitat modelled as upstream but stream widths of 3m vs 15m at the crossing. The potential for restoration of these systems may appear equivalent when only the linear length of stream is reported but when area is reported the amount of potential habitat upstream of the point on the 15m wide system is 5 times greater.

Some consideration of how this would play with calculating estimates of streamline width through lakes/wetlands/reservoirs or including all or a portion of those freshwater atlas generated areas will be necessary too.

**Comment by smnorris** [comment]
A potential method for calculating area based on the modelled channel width: https://gis.stackexchange.com/questions/340968/varying-size-buffer-along-line-with-postgis

**Comment by NewGraphEnvironment** [comment]
As a rough estimate we could use `length_m` * `channel_width` from `bcfishpass.streams`

**Comment by smnorris** [comment]
Good point. Variable width buffering might be neat for visualization but unnecessary for calculations.

**Comment by NewGraphEnvironment** [comment]
I wonder if now is a good time to add this? Thinking maybe just keeping it simple with `length_m` * `channel_width` from `bcfishpass.streams` should allow tallies quite easily afterwards

**Comment by smnorris** [comment]
New BT rearing model includes lakes and wetlands.

For translating linear reporting into area based reporting and potentially modifying the rearing waterbody requirements for CH/ST, I think a discussion with all users would be valuable. @captianmarmot @nickw-CWF are likely to have input.  

Possible issues for discussion:
- how to define area in wetlands/lakes in reporting - use the modelled CW for wetlands as suggested by @NewGraphEnvironment ?
- how might translating the linear numbers into areas relate to the lateral habitat modelling?
